source/common/ssl/context_impl.cc

```diff
@@ -223,6 +223,15 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
   return verified;
 }
 
+bool ContextImpl::uriMatch(const std::string& uriPattern, const char* uri) {
+  size_t pattern_len = uriPattern.length();
+  if (pattern_len > 1 && uriPattern[pattern_len - 1] == '*' && uriPattern[pattern_len - 2] == '/') {
+    return uriPattern.compare(0, pattern_len - 1, uri, pattern_len - 1) == 0;
```

Comment:
That's because the wrong method is used here. It should be:

    uriPattern.compare(0, pattern_len - 1, uri)

for NULL-terminated string, instead of:

    uriPattern.compare(0, pattern_len - 1, uri, pattern_len - 1)

which specifies length of the `uri` string.